frontstage/views/account/account.py

```diff
@@ -31,8 +38,42 @@ def update_account(session):
     if not form_valid:
         flash('At least one option should be selected.')
         return redirect(url_for('account_bp.get_account'))
-    if form.data['option'] == 'contact_details':
-        return redirect(url_for('account_bp.change_account_details'))
+    else:
+        return redirect(url_for(form_redirect_mapper.get(form.data['option'])))
+
+
+@account_bp.route('/change-password', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def change_password(session):
+    form = ChangePasswordFrom(request.values)
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
+    if request.method == 'POST' and form.validate():
+        username = respondent_details['emailAddress']
+        password = request.form.get('password')
+        new_password = request.form.get('new_password')
+        if new_password == password:
+            return render_template('account/account-change-password.html', form=form,
+                                   errors={"new_password": ["Your new password is the same as your old password"]})
+        bound_logger = logger.bind(email=obfuscate_email(username))
+        bound_logger.info("Attempting to find user in auth service")
+        try:
+            auth_controller.sign_in(username, password)
+            bound_logger.info("Attempting to change password via party service")
+            party_controller.change_password(username, new_password)
+            bound_logger.info("password changed via party service")
+            flash('Your password has been changed.')
+            return redirect(url_for('surveys_bp.get_survey_list', tag='todo'))
+        except AuthError as exc:
+            error_message = exc.auth_error
+            if BAD_AUTH_ERROR in error_message:
```

Comment:
Can you change this variable to `BAD_CREDENTIALS_ERROR` because that's what it is, and calling it this would make it clearer

Reply:
ok will change it, but I don't think `BAD_AUTH_ERROR` is bad or changing will make it clear :-)